openapi3/swagger_loader_http_error_test.go

```diff
@@ -0,0 +1,49 @@
+package openapi3
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/jarcoal/httpmock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLoadFromRemoteURLFailsWithHttpError(t *testing.T) {
+	spec := []byte(`
+{
+    "openapi": "3.0.0",
+    "info": {
+        "title": "",
+        "version": "1"
+    },
+    "paths": {
+        "/test": {
+            "post": {
+                "responses": {
+                    "default": {
+                        "description": "test",
+                        "headers": {
+                            "X-TEST-HEADER": {
+                                "$ref": "http://` + addr + `/components.openapi.json#/components/headers/CustomTestHeader"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}`)
+
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	httpmock.RegisterResponder("GET", "http://localhost:7965/components.openapi.json",
+		httpmock.NewStringResponder(400, ""))
+	loader := NewSwaggerLoader()
+	loader.IsExternalRefsAllowed = true
+	swagger, err := loader.LoadSwaggerFromDataWithPath(spec, &url.URL{Path: "testdata/testfilename.openapi.json"})
```

Comment:
Could you also test `LoadSwaggerFromData`? Reuse the `loader` please.

Reply:
I'll take a look at this.